### DISCUSSION TITLE
ask for help

### DISCUSSION THREAD

**Discussion by Vjze** [PyO3/pyo3#2809] [Questions]
I have a piece of python code that can be run directly without error, but it will report an error when called in rust. Maybe someone can help me, thank you.
![2](https://user-images.githubusercontent.com/99671889/208050572-9dbddc87-f525-42db-b679-c4869c3dcd3c.png)
![1](https://user-images.githubusercontent.com/99671889/208050588-c5a84e13-a588-45fe-8171-d3d8f9a47970.png)

**Comment by birkenfeld** [comment]
Since it's a "file not found" error, maybe the current directory is relevant?

Otherwise, since you're calling CLR stuff and the error comes from there, is there any important environment variables that might be different?

**Reply by Vjze** [reply]
They are all in the same directory, and what crl gets is the current directory. I won’t have any exceptions when I run print.py alone, but an error will be reported when I call print.py in rust.
![image](https://user-images.githubusercontent.com/99671889/208061568-c5944b98-fd6e-4b07-bc41-5d0d1806d972.png)

**Reply by Vjze** [reply]
If it is a problem with the current directory, then I should also report an error when I run print.py, I think so.

**Reply by adamreichold** [reply]
When you execute `cargo run`, the current directory is that of your project whereas the `debugpy/launcher` executable most likely changes the current directory to the `src` subdirectory where the `print.py` file is located. (Otherwise, you would probably need to do `sys.path.append(os.path.join(os.getcwd(), "src"))` to reach those DLL files.)

**Reply by Vjze** [reply]
Thanks for the hint, but now there is a new problem, my english is very poor, hope you can understand.
![屏幕截图 2022-12-17 125501](https://user-images.githubusercontent.com/99671889/208225957-77a00f38-2f1c-43c9-bb51-cbb1059fc36c.png)
